run

```diff
@@ -197,13 +197,35 @@ run() {
 	elif [[ ${FILE#*.} == go ]]; then
 		if [[ $COMPILE == true ]]; then
 			go build -o "${FILE%%.*}.out" "$FILE"
-			if [[ $PERSISTENT != true ]]; then
-				rm "${FILE%%.*}.out"
-			fi
 		else
 			go run "$FILE"
 		fi
+		if [[ $PERSISTENT != true ]]; then
+			rm "${FILE%%.*}.out"
+		fi
```

Comment:
Not much of a difference logic wise but maintains uniformity throughout the code - Nice